### ISSUE TITLE
Add a Sidebar widget to Quilter

### ISSUE THREAD

**Issue by lainsce** [lainsce/quilter#143]
Quilter needs a sidebar with recent files & Markdown symbol tree for added controls

**Comment by gnacho** [comment]
If you need a mockup or any help with design let me know, I'll be there :)

**Comment by lainsce** [comment]
I have a couple of mockups already, but they could be better tbh
![Recent Files](https://my.mixtape.moe/jllqts.png)
![Markdown Structure](https://my.mixtape.moe/vkthtg.png)

**Comment by gnacho** [comment]
Take a look at this: https://github.com/gnacho/Gnome-MD/blob/master/Main.png

Let me know if there's anything you'd like to add or remove.

**Comment by lainsce** [comment]
Yeah, I didn't like the vertical stacking of both sidebar functions, so I used a tab interface in my mockup.
Also, the whole sidebar would hide by pushing a button in the statusbar.

**Comment by gnacho** [comment]
Well, is the only way to have both features at the same time, as vertical space should be more than enough.

And TOC is not really useful most of the times, only for large documents.

Not sure if you want to use sidebar for open documents as [Marker](https://github.com/fabiocolacio/Marker) does or more like Gedit, which is a shorcurt for recent documents. I guess first one is more useful considering "new documents" as "Untitled #1" or something like that.

Also should be useful to add the path, as I can have a document using the same name under 2016, 2017, 2018 folders (just an example).

**Comment by lainsce** [comment]
Yeah, I think I'll format the recent documents list to show at least a "subtitle" if both documents are named the same, with the path.

**Comment by gnacho** [comment]
If it's for the recent documents you can show an open menu like Gedit does (it's pretty much the same style). I believe tabs feature could be more interesting tho.

I'm not sure about Elementary HIG, I'm using GNOME where looks great there too. You can check it here: https://developer.gnome.org/hig/stable/

**Comment by lainsce** [comment]
Yeah, my apps only use the elementary HIG, not GNOME's, so one thing that works in GNOME's might not work in elementary's

**Comment by lainsce** [comment]
Unintentional side-effect of not using custom styling in the headerbar. :smile:

**Comment by surajmandalcell** [comment]
Isnt quilter supposed to be minimal editor which lets you focus on writing?
Why a sidebar? it isn't a code editor right? (or is it?) A better choice would be a resents icon in toolbar which shows the recent folders and files opened. 
If you want i can do a mockup to show how that looks.

**Comment by lainsce** [comment]
The sidebar won't be always showing all the time, I'm planning on making it hide-able through a button on the status bar, and as default, the sidebar will be set as don't show. Don't worry, I'm not going to make it not minimal. Besides, I think that having a sidebar will make my app look more inviting to users of Ulysses and iA Writer, both also minimal story writing apps on other platforms, and also having the sidebar approach to showing recents.

**Comment by surajmandalcell** [comment]
ok, if it looks good then sure!

**Comment by infosisio** [comment]
@lainsce I was about to post a request for this. Is it a feature about to come ?

**Comment by lainsce** [comment]
It's kind of a feature on hold. :disappointed_relieved:

**Comment by BharatKalluri** [comment]
Hello, is there plans on adding support to open a folders in quilter?

**Comment by lainsce** [comment]
Opening a whole folder and displaying its contents no, but saving the 5 or so most recent files and displaying them in this Sidebar is.

**Comment by lainsce** [comment]
This is FINALLY being implementeranno! Questa!

**Comment by lainsce** [comment]
![shot](https://user-images.githubusercontent.com/4886639/47199463-2237fa00-d348-11e8-8827-622aa5fdeeaf.png)
:tada:
